### PR TITLE
Fix archName after encoding/decoding

### PR DESCRIPTION
**Pull request by MatkovIvan**
* [ ] Has `CHANGELOG.md` been updated?
* [x] Are tests passing locally?
* [x] Are the files formatted correctly?
* [ ] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Fix archName after encoding/decoding

## Related PRs or issues

[AB#94002](https://dev.azure.com/msmobilecenter/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/94002)

### CODE REVIEW

File: AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m [hunk 1]
```diff
@@ -315,14 +312,8 @@ + (MSACErrorReport *)errorReportFromLog:(MSACAppleErrorLog *)errorLog {
...
   NSDate *appErrorTime = errorLog.timestamp;
-  NSString *codeType = unknownString;
-  NSString *archName = unknownString;
-  if (errorLog.isKnownEncodingType && errorLog.primaryArchitectureId != nil) {
-    codeType = [self convertCodeTypeToString:errorLog.primaryArchitectureId.longValue];
-    if (errorLog.architectureVariantId != nil) {
-      archName = [self convertArchNameToString:errorLog.primaryArchitectureId.longValue subtype:errorLog.architectureVariantId.intValue];
-    }
-  }
+  NSString *codeType = [self convertCodeTypeToString:errorLog.primaryArchitectureId.longValue];
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m:315]: Why don't we check `if (primaryArchitectureId != nil)` and `if (errorLog.architectureVariantId != nil)` anymore?
Also we are not setting `codeType = unknownString` anymore. Is it expected?

File: AppCenterCrashes/AppCenterCrashesTests/MSACErrorLogFormatterTests.mm [hunk 2]
```diff
@@ -766,97 +764,41 @@ - (void)checkIfArchNameConvertedCorrectly:(NSNumber *)codeType archName:(NSNumbe
...
 
-- (void)testConvertCodeTypeUnknownEncodingTypeToString {
-  [self checkIfCodeTypeConvertedCorrectly:@(CPU_TYPE_ARM) isKnownEncodingType:NO expectedType:@"???"];
-  [self checkIfCodeTypeConvertedCorrectly:@(CPU_TYPE_ARM64) isKnownEncodingType:NO expectedType:@"???"];
-  [self checkIfCodeTypeConvertedCorrectly:@(CPU_TYPE_X86) isKnownEncodingType:NO expectedType:@"???"];
-  [self checkIfCodeTypeConvertedCorrectly:@(CPU_TYPE_X86_64) isKnownEncodingType:NO expectedType:@"???"];
-  [self checkIfCodeTypeConvertedCorrectly:@(CPU_TYPE_POWERPC) isKnownEncodingType:NO expectedType:@"???"];
-}
-
-- (void)testConvertArchNameUnknownEncodingTypeToString {
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashesTests/MSACErrorLogFormatterTests.mm:823]: It is still possible that archName will be set to "???" if `archName` (first parameter in `convertArchNameToString`) was not matched to the list in `switch`. Shouldn't we cover this scenario?

File: AppCenterCrashes/AppCenterCrashesTests/MSACErrorLogFormatterTests.mm [hunk 3]
```diff
@@ -766,97 +764,41 @@ - (void)checkIfArchNameConvertedCorrectly:(NSNumber *)codeType archName:(NSNumbe
...
                              archName:@(CPU_TYPE_X86_64)
-                  isKnownEncodingType:YES
                          expectedName:@"x86_64"];
   [self checkIfArchNameConvertedCorrectly:@(CPU_TYPE_POWERPC)
                              archName:@(CPU_TYPE_POWERPC)
-                  isKnownEncodingType:YES
                          expectedName:@"powerpc"];
 }
 
-- (void)testConvertCodeTypeUnknownEncodingTypeToString {
```
Comment (DmitriyKirakosyan) [inline AppCenterCrashes/AppCenterCrashesTests/MSACErrorLogFormatterTests.mm:815]: What about the case when archName is not found? This line https://github.com/microsoft/appcenter-sdk-apple/blob/5105fc4ff7dba9238546a170b090ef4b0ac2415e/AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m#L473.

File: AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m (hunk 1 above)
Comment (MikhailSuendukov) [reply AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m]: rolled back this check

Comment (MatkovIvan) [reply AppCenterCrashes/AppCenterCrashes/Internals/Util/MSACErrorLogFormatter.m]: It WAS expected. It's handled inside `convertCodeTypeToString` and `convertArchNameToString:errorLog` functions

**Review by DmitriyKirakosyan — APPROVED** [review]

**Pull request merged**